### DISCUSSION TITLE
How to get voice messages with telegraf v4.x ?

### DISCUSSION THREAD

**Discussion by marcopiii** [telegraf/telegraf#1432] [Q&A]
If a user sends a voice message to my bot, I would like to be able to do something with it, for example, save its `id` somewhere.

I remember that with version `3.x`, having the `ContextMessageUpdate` object I could do `ctx.message.voice.file_id`, but it seems that in `v4.x` the `message` object in the `Context` has no such `voice` field. I can't find any useful documentation about it.

Can someone give me a hint?

**Comment by wojpawlik** [comment]
https://github.com/telegraf/telegraf/issues/1388#issuecomment-791573609
